src/androidTestUnbrandedWebviewDebug/java/org/medicmobile/webapp/mobile/LastUrlTest.java

```diff
@@ -0,0 +1,79 @@
+package org.medicmobile.webapp.mobile;
+
+import static androidx.test.espresso.Espresso.onData;
+import static androidx.test.espresso.Espresso.onView;
+import static androidx.test.espresso.action.ViewActions.click;
+import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
+import static androidx.test.espresso.action.ViewActions.replaceText;
+import static androidx.test.espresso.matcher.ViewMatchers.withId;
+import static androidx.test.espresso.matcher.ViewMatchers.withText;
+import static androidx.test.espresso.web.assertion.WebViewAssertions.webMatches;
+import static androidx.test.espresso.web.sugar.Web.onWebView;
+import static androidx.test.espresso.web.webdriver.DriverAtoms.findElement;
+import static androidx.test.espresso.web.webdriver.DriverAtoms.getText;
+import static androidx.test.espresso.web.webdriver.DriverAtoms.webClick;
+
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.anything;
+import static org.hamcrest.Matchers.containsString;
+
+import androidx.test.espresso.DataInteraction;
+import androidx.test.espresso.ViewInteraction;
+import androidx.test.espresso.web.webdriver.Locator;
+import androidx.test.ext.junit.rules.ActivityScenarioRule;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+import androidx.test.filters.LargeTest;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+
+/**
+ * Test that when the app is closed and then opened again, the last URL
+ * viewed is loaded instead of the startup URL.
+ */
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class LastUrlTest {
+
+	@Rule
+	public ActivityScenarioRule<SettingsDialogActivity> mActivityTestRule =
+			new ActivityScenarioRule<>(SettingsDialogActivity.class);
+
+	@Test
+	public void somethingToTest() throws InterruptedException {
+		onView(withText("Custom")).perform(click());
+		ViewInteraction textAppUrl = onView(withId(R.id.txtAppUrl));
+		textAppUrl.perform(replaceText("https://angular.io/resources?category=development"),
+				closeSoftKeyboard());
+		onView(withId(R.id.btnSaveSettings)).perform(click());
+		Thread.sleep(2000);
+		// Click into "Education" to go to "?category=education"
+		onWebView()
+				.withNoTimeout()
+				.withElement(findElement(Locator.XPATH, "//a[contains(text(), 'Education')]"))
+				.perform(webClick());
```

Comment:
We probably don't need to click on a new page, right? If the app remembers the original angular.io URL then the code is working.

Reply:
The first interaction with the UI is to set the CHT URL that is stored only once in the app storage (Unbranded flavor), and after that we need to go to a different URL to verify that when the app is reopened, instead of loading the CHT URL, it loads the last URL opened.

Reply:
Ahh right. Makes sense. Thanks for the explanation!